src/main.py

```diff
@@ -276,13 +276,14 @@ def load_supported_services(context: LoggingContext, selected_featuresets: List[
                     service_name=service_yaml.get("service", "None")
                     featureSet=service_yaml.get("featureSet", "None")
                     # If whitelist of services exists and current service is not present in it, skip
+                    whitelist_exists = selected_featuresets is not None and selected_featuresets.__len__() > 0
                     if activation_config:
                         activation_config_feature_sets = activation_config.get(service_name, {}).get("featureSets", [])
-                        should_skip = featureSet not in activation_config_feature_sets
+                        should_skip = whitelist_exists and featureSet not in activation_config_feature_sets
```

Comment:
selected_featuresets should be used only as a fallback (in else section, probably I should have updated the comment) - it's a previous solution (in the config we could specify list of gcp services to monitor).
Currently with new deployments we check if feature set from service config is present in activation_config, if not then skip it.